### DISCUSSION TITLE
💼 Copilot Enterprise is now generally available

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#110324] [Copilot News and Announcements]
![Screenshot 2024-02-27 at 5 39 45 PM](https://github.com/community/community/assets/95834130/49f4f904-4afe-4d6e-a071-87034210bc3f)
> Alt Text: Logo for GitHub Copilot, featuring a sleek design with a modern color palette and bold typography.

Copilot Enterprise, our most advanced AI offering to date, [is now generally available](https://github.blog/2024-02-27-github-copilot-enterprise-is-now-generally-available). With Copilot Enterprise, you can:

*  ⚡ **Gain a deeper understanding of your organization’s unique codebase**: Copilot Chat in GitHub.com understands your code and streamlines code navigation and comprehension for developers.
*  ✨ **Quickly access organizational knowledge and best practices**: By letting developers attach knowledge bases (formerly known as docsets) to conversations, Copilot Chat in GitHub.com can answer questions based on your Markdown documentation stored on GitHub.
*  ⏩  **Review pull requests faster**: With pull request summaries generated by Copilot and the ability to chat about changes in a pull request, reviewers can get up to speed on a pull request quickly and spend more time providing valuable feedback.

> [Learn more about what’s included with GitHub Copilot Enterprise >](http://github.com/features/copilot#pricing)

Following on from our [limited public beta](https://github.blog/2023-11-08-universe-2023-copilot-transforms-github-into-the-ai-powered-developer-platform/#introducing-github-copilot-enterprise-copilot-personalized-to-your-organization), we are bringing the following improvements to Copilot Enterprise today to make Copilot even smarter:

*  1️⃣  **Copilot can now search Bing within chat conversations in GitHub.com** to answer questions and find information outside of its general knowledge or your codebase _(public beta)_.
*  2️⃣  **You can now access your knowledge bases (formerly known as docsets) from any Copilot Chat conversation in GitHub.com** with the “Attach knowledge” button. Organization owners can [create knowledge bases](https://docs.github.com/enterprise-cloud@latest/copilot/github-copilot-enterprise/copilot-chat-in-github/managing-copilot-knowledge-bases) from an organization’s settings.
*  3️⃣  **Copilot knows about code as you browse**, so you no longer have to be explicit about exactly what file, symbol or snippet you want to chat about.

![Example conversation demonstrating how GitHub Copilot can access the code you are currently looking at](https://github.com/github/release-assets/assets/884151/39816e1c-5545-431a-924f-639cd3b6ccfa)

*  4️⃣  **Copilot generates pull summaries that are now more structured**, with a “Summary” section that gives a high-level overview, and an “Outline” section that walks through the code.
*  5️⃣  **Copilot can now analyze and explain any pull request diff**, making it easier for pull request reviewers to understand changes and share great feedback.

![Example conversation demonstrating how GitHub Copilot can explain and improve pull request diffs](https://github.com/github/release-assets/assets/884151/81ae9bff-1844-4b8f-b501-ad3abdb78f06)

> [Get started with Copilot Enterprise](https://docs.github.com/en/copilot/github-copilot-enterprise/overview/about-github-copilot-enterprise)

Ready to give Copilot Chat in GitHub.com a try? Here are some suggested prompts to get you started:

*   **Ask a question about recent events to trigger a Bing search**: What updates were there in Node.js v20?
*   **Open Copilot Chat on a repository and ask a question about the repository**: Where is the turnOn function defined?
*   **Open a file on GitHub.com and ask a question about that file**: Draft unit test cases for each of the functions in the file I’m currently viewing.

Our most advanced AI offering to date is customized to your organization’s knowledge and codebase, infusing GitHub Copilot throughout the software development lifecycle. Learn more about the Copilot Enterprise in this [blog post](https://github.blog/2024-02-27-github-copilot-enterprise-is-now-generally-available/). 🚀

**Comment by gmathi2** [comment]
Are those features only available when using Copilot Chat in GitHub.com or also in my IDE of choice?

**Reply by AriMKatz** [reply]
Seconding this question

**Reply by adrianegraphene** [reply]
Based off this video, the features seem to be available through the IDE extensions, but don't quote me on that. 
https://youtu.be/vUX5u_4B2AM?t=581

**Reply by gmathi2** [reply]
@adrianegraphene 
The things shown in the IDE in your video are nothing new and were already available in the other Copilot versions

**Reply by Akash1134** [reply]
Hi @gmathi2 , Thanks for dropping in your question :) The features launched with [Copilot Enterprise](https://github.blog/2024-02-27-github-copilot-enterprise-is-now-generally-available/) are currently specific to Copilot Chat experience in GitHub.com.

**Reply by gmathi2** [reply]
@Akash1134 I suppose this will also apply to the announced "fine-tuned models"? No tuning on repos not hosted on GitHub.com?

**Reply by AriMKatz** [reply]
Hi @Akash1134 so no access to personalized codebase/doc rag in vscode????? That's the only feature I care about and is frankly critical

**Reply by gmathi2** [reply]
@AriMKatz Maybe they will find a way to give you access to a fine tuned model based on your codebase in VS code, but I'm afraid they won't find a way to tune a model if your codebase is not hosted on GitHub.com.

**Reply by AriMKatz** [reply]
I don't mean a finetuned model. I'm talking about personalized chat based on rag from indexed codebases /knowledge stores

**Comment by omid-ebrimi** [comment]
Nice

**Comment by rickywck** [comment]
For the feature "Gain a deeper understanding of your organization’s unique codebase", is the codebase has to be stored in GitHub Enterprise? So, for companies which are using other code repository tools, they can't have access to this feature? Btw, I can't find any information on this feature from the documentation published so far.
